.github/workflows/flutter.yml

```diff
@@ -0,0 +1,34 @@
+name: CI
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches:
+      - master
+      - feature/*
+      - bugfix/*
+      - hotfix/*
+
+jobs:
+  lint:
```

Comment:
I think this does more than linting, so its probably better to rename this to encapsulate all functionality:
- quality_check
- code evaluation
or a better name if you come up with one

Or separate the test and linting steps in different jobs